llvm/lib/CodeGen/PatchableFunction.cpp

```diff
@@ -87,6 +87,9 @@ bool PatchableFunction::runOnMachineFunction(MachineFunction &MF) {
   for (auto &MO : FirstActualI->operands())
     MIB.add(MO);
 
+  if (FirstActualI->shouldUpdateCallSiteInfo())
+    MF.eraseCallSiteInfo(&*FirstActualI);
```

Comment:
Maybe we don't need this. According to the description above, `PatchableFunction` needs the first instruction is two-byte long.

http://xxeo.com/single-byte-or-small-x86-opcodes

One-byte instruction is rare in X86. `TailCall` can be transformed to either  a `call` or a `jmp`, which is always longer than 2 bytes. I think we can just skip emitting patchable-op if the first instruction is a `TailCall`.